e2etester/createtestworkflows.go

```diff
@@ -0,0 +1,243 @@
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+// Package e2etester is a library for VM-based e2e tests using Daisy workflows.
+package e2etester
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"reflect"
+	"regexp"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+	"google.golang.org/api/compute/v1"
+)
+
+// Config is a config.
+type Config struct {
+	Image    string
+	Metadata map[string]string
+}
+
+var configs map[string]Config
+var dependencies map[string]string
+
+// RunTogether runs two tests together by including the dependee into the
+// dependent WF.
+func RunTogether(dependent, dependee string) {
+	if dependencies == nil {
+		dependencies = make(map[string]string)
+	}
+	dependencies[dependent] = dependee
+}
+
+// AddTestConfig adds a Config object for a specified test.
+func AddTestConfig(test string, config Config) {
+	if configs == nil {
+		configs = make(map[string]Config)
+	}
+	configs[test] = config
+}
+
+var md map[string]string
+
+// GetMetadata gets instance attributes from metadata.
+func GetMetadata() (map[string]string, error) {
+	if md != nil {
+		return md, nil
+	}
+	metadataURL := "http://metadata.google.internal/computeMetadata/v1/instance/attributes/?recursive=true&alt=json"
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", metadataURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error building HTTP request for metadata: %s", err)
+	}
+	req.Header.Add("Metadata-Flavor", "Google")
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("error getting metadata: %s", err)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
+	if err != nil {
+		return nil, fmt.Errorf("error getting metadata: %s", err)
+	}
+	err = json.Unmarshal(body, &md)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing metadata: %s", err)
+	}
+	return md, nil
+}
+
+func runLocalTest(tests interface{}) error {
+	md, err := GetMetadata()
+	if err != nil {
+		return err
+	}
+	testname, ok := md["TestName"]
+	if !ok {
+		return fmt.Errorf("TestName not in metadata")
+	}
+
+	typ := reflect.TypeOf(tests)
+	m, ok := typ.MethodByName(testname)
+	if !ok {
+		return fmt.Errorf("could not locate specified test %q", testname)
+	}
+	if m.Type.NumIn() != 1 {
+		return fmt.Errorf("func %q should only have one input", testname)
+	}
+	if m.Type.In(0) != typ {
+		return fmt.Errorf("func %q should take %v type as first arg, but takes %v", testname, typ, m.Type.In(0))
```

Comment:
Is the one expected arg actually the test object func is associate with ("this"), or something else?